crates/assistant2/src/context_store.rs

```diff
@@ -44,4 +45,13 @@ impl ContextStore {
     pub fn remove_context(&mut self, id: &ContextId) {
         self.context.retain(|context| context.id != *id);
     }
+
+    pub fn contains_project_entry(&self, entry_id: ProjectEntryId) -> bool {
+        self.context.iter().any(|probe| match probe.kind {
+            ContextKind::File(probe_entry_id) => probe_entry_id == entry_id,
+            ContextKind::Directory => false,
+            ContextKind::FetchedUrl => false,
+            ContextKind::Thread => false,
+        })
+    }
```

Comment:
I think the linear search should be ok for the current usage, but we may want to consider an alternative approach if we use this to filter `file_context_picker` entries later.